packages/insured-bridge-relayer/src/canonical-bridge-adapters/ArbitrumBridgeAdapter.ts

```diff
@@ -0,0 +1,121 @@
+import winston from "winston";
+import { ZERO_ADDRESS } from "@uma/common";
+
+import { providers, VoidSigner } from "ethers";
+import { Bridge, OutgoingMessageState } from "arb-ts";
+
+import { Outbox__factory } from "arb-ts/dist/lib/abi/factories/Outbox__factory";
+
+import BridgeAdapterInterface from "./BridgeAdapterInterface";
+
+import Web3 from "web3";
+import type { TransactionType } from "@uma/common";
+
+export class ArbitrumBridgeAdapter implements BridgeAdapterInterface {
+  public bridge: Bridge | undefined = undefined;
+
+  constructor(readonly logger: winston.Logger, readonly l1Web3: Web3, readonly l2Web3: Web3) {}
+
+  async initialize() {
+    const l1EthersProvider = new providers.Web3Provider(this.l1Web3.currentProvider as any);
+    const l1EthersVoidSigner = new VoidSigner(ZERO_ADDRESS, l1EthersProvider);
+    const l2EthersProvider = new providers.Web3Provider(this.l2Web3.currentProvider as any);
+    const l2EthersVoidSigner = new VoidSigner(ZERO_ADDRESS, l2EthersProvider);
+
+    this.bridge = await Bridge.init(l1EthersVoidSigner, l2EthersVoidSigner);
+
+    this.logger.debug({ at: "ArbitrumBridgeAdapter", message: "Initialized Arbitrum Bridge Adapter" });
```

Comment:
Note: if these methods + _throwIfNotInitialized() checks are annoying, one thing you could do is use async factory functions to create classes with async logic.

Reply:
good call out. I ended up just moving the check into the `constructCrossDomainFinalizationTransaction`  method seeing this the only place we use this.